### PR TITLE
suite-nonsplit-returns

### PR DESCRIPTION
**Pull request by gorav-sharma-metacube**
## PR Description
CC-16840: suite-nonsplit-returns

### CODE REVIEW

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+Resource    ../../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+ ####POST####
+Create_a_return_with_Invalid_access_token
+     [Setup]    I set Headers:    Authorization=3485h7
+    When I send a POST request:     /returns     {"data":{"type":"returns","attributes":{"store":"DE","returnItems":[{"salesOrderItemUuid":"${random}","reason":"${return_reason_damaged}"}]}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot:14]: Please use store value as a variable

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    Then Response status code should be:     401
+    And Response reason should be:     Unauthorized
+    And Response should return error message:    Invalid access token.
+    And Response should return error code:    001
+
+Create_a_return_with_without_access_token
+     [Setup]    I set Headers:    Authorization=
+    When I send a POST request:     /returns     {"data":{"type":"returns","attributes":{"store":"DE","returnItems":[{"salesOrderItemUuid":"${random}","reason":"${return_reason_damaged}"}]}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot:22]: Please use store value as a variable

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    ...    AND    Save value to a variable:    [included][0][attributes][items][0][uuid]    Uuid
+    When I send a POST request:     /returns     {"data":{"type":"returns","attributes":{"store":"DE","returnItems":[{"salesOrderItemUuid":"${Uuid}","reason":"${return_reason_damaged}"}]}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot:37]: Please use store value as a variable

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    When I send a POST request:     /returns     {"data":{"type":"returns","attributes":{"store":"${store.de}","returnItems":[{"salesOrderItemUuid":"${returnableSalesOrderItemUuid}","reason":"${return_reason_damaged}"}]}}}
+    Then Response status code should be:     422
+    And Response reason should be:     Unprocessable Content
+    And Response should return error message:    "Return cant be created."    
+    And Response should return error code:    3601
+
+Create_return_with_invalid_returnItems_uuid
+     [Setup]    Run Keywords    I get access token for the customer:    ${yves_second_user.email}
+    ...    AND    I set Headers:    Authorization=${token}
+    When I send a POST request:     /returns     {"data":{"type":"returns","attributes":{"store":"DE","returnItems":[{"salesOrderItemUuid":"${random}","reason":"${return_reason_damaged}"}]}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot:60]: Please use store value as a variable

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    Then Response status code should be:     422
+    And Response reason should be:     Unprocessable Content
+    And Response should return error message:    "Return cant be created."    
+    And Response should return error code:    3601
+
+Create_return_without_returnItems_uuid
+     [Setup]    Run Keywords    I get access token for the customer:    ${yves_second_user.email}
+    ...    AND    I set Headers:    Authorization=${token}
+    When I send a POST request:     /returns     {"data":{"type":"returns","attributes":{"store":"DE","returnItems":[{"salesOrderItemUuid":"","reason":"${return_reason_damaged}"}]}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot:69]: Please use store value as a variable

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    Then Response status code should be:     422
+    And Response reason should be:     Unprocessable Content
+    And Response should return error message:    "Return cant be created."    
+    And Response should return error code:    3601
+
+Create_return_without_returnItems_reason
+     [Setup]    Run Keywords    I get access token for the customer:    ${yves_second_user.email}
+    ...    AND    I set Headers:    Authorization=${token}
+    When I send a POST request:     /returns     {"data":{"type":"returns","attributes":{"store":"DE","returnItems":[{"salesOrderItemUuid":"920fc22f-3fb6-53ec-bc5e-c4d321115462","reason":""}]}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot:78]: Please use store value as a variable

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/positive.robot [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+    ...    AND    Save value to a variable:    [included][0][attributes][items][0][uuid]    Uuid
+    ...    AND    Save value to a variable:    [included][0][attributes][items][0][refundableAmount]    productPrice
+    ...    AND    Update order status in Database:    shipped
+    
+    When I send a POST request:     /returns     {"data":{"type":"returns","attributes":{"store":"DE","returnItems":[{"salesOrderItemUuid":"${Uuid}","reason":"${return_reason_damaged}"}]}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/return_endpoints/returns/positive.robot:24]: Please use store value as a variable

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/positive.robot [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+    ...    AND    Save value to a variable:    [included][0][attributes][items][0][uuid]    Uuid
+    ...    AND    Save value to a variable:    [included][0][attributes][items][0][refundableAmount]    productPrice
+    ...    AND    Update order status in Database:    shipped
+
+    When I send a POST request:     /returns?include=return-items     {"data":{"type":"returns","attributes":{"store":"DE","returnItems":[{"salesOrderItemUuid":"${Uuid}","reason":"${return_reason_damaged}"}]}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/return_endpoints/returns/positive.robot:49]: Please use store value as a variable here and in other tests below

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot (hunk 1 above)
Comment (gorav-sharma-metacube) [reply tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot]: fixed

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot (hunk 2 above)
Comment (gorav-sharma-metacube) [reply tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot]: fixed

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot (hunk 3 above)
Comment (gorav-sharma-metacube) [reply tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot]: fixed

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot (hunk 5 above)
Comment (gorav-sharma-metacube) [reply tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot]: fixed

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot (hunk 6 above)
Comment (gorav-sharma-metacube) [reply tests/api/suite_nonsplit/glue/return_endpoints/returns/negative.robot]: fixed

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/positive.robot (hunk 7 above)
Comment (gorav-sharma-metacube) [reply tests/api/suite_nonsplit/glue/return_endpoints/returns/positive.robot]: fixed

File: tests/api/suite_nonsplit/glue/return_endpoints/returns/positive.robot (hunk 8 above)
Comment (gorav-sharma-metacube) [reply tests/api/suite_nonsplit/glue/return_endpoints/returns/positive.robot]: fixed

**Review by Katerynaa-Lavryk — DISMISSED** [review]

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**